### ISSUE TITLE
Custom transformer fails to compile with Scala 2.11

### ISSUE THREAD

**Issue by epkjoja** [scalalandio/chimney#124]
I am trying to write a custom, nested transformer and it works beautifully when using Scala 2.12 but it fails during compilation when using Scala 2.11.12. I don't think it's a specific Chimney problem but since I'm unable to figure out a solution (and others might be facing the same problem), I'm opening an issue here.

```scala
  implicit def optTransformer[A, B]
      (implicit trans: Transformer[A, B]): Transformer[Option[A], Option[B]] =
    (opt: Option[A]) =>
      opt match {
        case Some(value) => Some(trans.transform(value))
        case None => None
      }
```

When compiling this with Scala 2.11.12 I get the following error:
```
[error] chimney/src/main/scala/Chimney.scala:10:22: type mismatch;
[error]  found   : Option[A] => Option[B]
[error]  required: io.scalaland.chimney.Transformer[Option[A],Option[B]]
[error]     (opt: Option[A]) =>
[error]                      ^
```

**Comment by krzemin** [comment]
You are using SAM (single abstract method) syntax which is not available for scala 2.11 by default. It's possible under `-Xexperimental` compiler flag.

Otherwise you can define your transformers with a bit more boilerplate anonymously implementing `Transformer` trait:

```
 implicit def optTransformer[A, B](...) = new Trasformer[...] {
  def transform(a: A): B = ...
}
```

which is supported by all Scala versions.

More info about SAM support in Scala:
- https://www.waitingforcode.com/scala-core/java-sam-scala/read
- https://herringtondarkholme.github.io/2015/01/24/scala-sam/

**Comment by epkjoja** [comment]
Thanks for the explanation! I added the experimental flag and that solved the issue.
